### PR TITLE
Drop disabled users

### PR DESCRIPTION
**Pull request by sethvargo**
We can use soft deletes with gorm to "delete" users. Fixes GH-153.

### CODE REVIEW

File: pkg/database/migrations.go [hunk 1]
```diff
@@ -342,6 +342,17 @@ func (db *Database) getMigrations(ctx context.Context) *gormigrate.Gormigrate {
...
 			},
 		},
+		{
+			ID: "00014-DropUserDisabled",
+			Migrate: func(tx *gorm.DB) error {
+				logger.Infof("db migrations: dropping user disabled column")
+				return tx.Model(&User{}).DropColumn("disabled").Error
+			},
+			Rollback: func(tx *gorm.DB) error {
+				sql := "ALTER TABLE users ADD COLUMN disabled bool NOT NULL DEFAULT true"
```
Comment (sethvargo) [inline pkg/database/migrations.go:352]: I could make this more complete, keying off of deleted_at, but it becomes a much more expensive operation.

File: pkg/database/migrations.go [hunk 2]
```diff
@@ -342,6 +342,18 @@ func (db *Database) getMigrations(ctx context.Context) *gormigrate.Gormigrate {
 				return nil
 			},
 		},
+		{
+			ID: "00014-DropUserDisabled",
+			Migrate: func(tx *gorm.DB) error {
+				logger.Infof("db migrations: dropping user disabled column")
+				sql := "ALTER TABLE users DROP COLUMN IF EXISTS disabled"
```
Comment (mikehelmick) [inline pkg/database/migrations.go:349]: Q: does this drop the column from the users_purge_index?

also, perhaps we don't need that index at all and should drop it here, if it exists.

File: pkg/database/migrations.go (hunk 1 above)
Comment (mikehelmick) [reply pkg/database/migrations.go:353]: this is fine.

File: pkg/database/migrations.go (hunk 2 above)
Comment (sethvargo) [reply pkg/database/migrations.go:360]: > When you remove a column from a table, PostgreSQL will automatically remove all of the indexes and constraints that involved the dropped column.

Seems so. Added a step to remove the index.

**Comment by sethvargo** [comment]
@mikehelmick PT(another)L

**Pull request merged**